frontend/public/components/storage-class-form.tsx

```diff
@@ -0,0 +1,927 @@
+            </Link>
+          </p>
+          <NameValueEditor
+            nameValuePairs={this.state.customParams}
+            nameString="Parameter"
+            valueString="Value"
+            addString="Add Parameter"
+            updateParentData={this.updateCustomParams} />
+        </FormGroup>
+      </React.Fragment>
+    );
+  };
+
+  close = () => {
+    this.props.stopK8sWatch(this.reduxId);
+    if (typeof this.props.onClose === 'function') {
+      this.props.onClose();
+    }
+    history.push('/k8s/cluster/storageclasses');
+  };
+
+  render() {
+    const newStorageClass = this.state.newStorageClass;
+    const reclaimPolicyKey = newStorageClass.reclaim === null ? this.reclaimPolicies.Delete : newStorageClass.reclaim;
+    const fieldErrors = this.state.fieldErrors;
+
+    return (
+      <div className="co-m-pane__body">
+        <Form className="wizard-form">
+          <h1 className="co-m-pane__heading">Create Storage Class</h1>
+
+          <FormGroup controlId={'basic-settings-name'} validationState={fieldErrors.nameValidationMsg ? 'error': null}>
+            <label className="control-label co-required" htmlFor="storage-class-name">Name</label>
+            <FormControl
+              type="text"
+              placeholder={newStorageClass.name}
+              id="storage-class-name"
+              // onChange={(event) => this.setNameHandler(event)}
+              onChange={(event) => this.setStorageHandler('name', event.target.value)}
+              value={_.get(newStorageClass, 'name', '')} />
+            <HelpBlock>{fieldErrors.nameValidationMsg ? fieldErrors.nameValidationMsg : null}</HelpBlock>
+          </FormGroup>
+
+          <FormGroup controlId={'basic-settings-description'}>
+            <label htmlFor="storage-class-description">Description</label>
+            <FormControl
+              type="text"
+              id="storage-class-description"
+              onChange={(event) => this.setStorageHandler('description', event.target.value)}
+              value={_.get(newStorageClass, 'description', '')} />
+          </FormGroup>
+
+          <FormGroup controlId={'basic-settings-reclaim-policy'}>
+            <label className="control-label co-required">Reclaim Policy</label>
+            <Dropdown
+              title="Choose Reclaim Policy"
+              items={this.reclaimPolicies}
+              dropDownClassName="dropdown--full-width"
+              selectedKey={reclaimPolicyKey}
+              onChange={(event) => this.setStorageHandler('reclaim', event)} />
+            <HelpBlock>
+              Determines what happens to persistent volumes when the associated persistent volume claim is deleted. Defaults to &lsquo;Delete&rsquo;
+            </HelpBlock>
+          </FormGroup>
+
+          <FormGroup controlId={'basic-settings-storage'}>
+            <label className="control-label co-required">Provisioner</label>
+            <Dropdown
```

Comment:
Would be nice if we enabled typeahead on this dropdown.

Reply:
Done